public/app/plugins/datasource/prometheus/querybuilder/state.ts

```diff
@@ -36,7 +36,7 @@ function getDefaultEditorMode(expr: string, defaultEditor: QueryEditorMode = Que
  * Returns query with defaults, and boolean true/false depending on change was required
  */
 export function getQueryWithDefaults(
-  query: PromQuery,
+  query: PromQuery & { expr?: string },
```

Comment:
I don't think we need this part. The comment line is clearly explaining the situation.